tasks/Task.cpp

```diff
@@ -13,37 +15,76 @@ Task::~Task()
 {
 }
 
+static RigidBodyState convertToRBS(PingStatus const& data)
+{
+    RigidBodyState rbs;
```

Comment:
It is a good practice to fill the time/timestamp field of the outputs...
if you don't have that information from the `pingStatus`, you can do `base::Time::now()`

Reply:
I'm going to add that it's not "a good practice", it is a required, super important one.